x-pack/plugins/cases/common/types/domain/case/v1.ts

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import * as rt from 'io-ts';
+import { CaseStatuses } from '@kbn/cases-components/src/status/types';
```

Comment:
Not specific to this PR but, why is this exported in `cases-components`?

Through the code I see instances of `import type { CaseStatuses } from '@kbn/cases-plugin/common';` and `import { CaseStatuses } from '@kbn/cases-components';`.

Reply:
The reason the `CaseStatuses` is in a package is that the package cannot import code from a plugin. We needed to create a component that uses `CaseStatuses` and put it on a package so it can be shared by other plugins. So we decided to move the enum into a package and cases and other plugins will import from them. The reason you see `import type { CaseStatuses } from '@kbn/cases-plugin/common';` is because somewhere in cases we do `export { CaseStatuses } from '@kbn/cases-components';`. I have a note to handle this on another PR.